src/Promitor.Integrations.AzureMonitor/AzureMonitorClient.cs

```diff
@@ -108,9 +109,16 @@ public AzureMonitorClient(AzureEnvironment azureCloud, string tenantId, string s
 
                 // Get the metric value according to the requested aggregation type
                 var requestedMetricAggregate = InterpretMetricValue(aggregationType, mostRecentMetricValue);
-
-                var measuredMetric = string.IsNullOrWhiteSpace(metricDimension) ? MeasuredMetric.CreateWithoutDimension(requestedMetricAggregate) : MeasuredMetric.CreateForDimension(requestedMetricAggregate, metricDimension, timeseries);
-                measuredMetrics.Add(measuredMetric);
+                try 
+                {
+                    var measuredMetric = string.IsNullOrWhiteSpace(metricDimension) ? MeasuredMetric.CreateWithoutDimension(requestedMetricAggregate) : MeasuredMetric.CreateForDimension(requestedMetricAggregate, metricDimension, timeseries);
```

Comment:
Formatting nit (I should have done earlier myself)
```suggestion
                    var measuredMetric = string.IsNullOrWhiteSpace(metricDimension)
                            ? MeasuredMetric.CreateWithoutDimension(requestedMetricAggregate)
                            : MeasuredMetric.CreateForDimension(requestedMetricAggregate, metricDimension, timeseries);
```

Reply:
Also fixed it in AzureMonitorScraper